src/shipstats.c

```diff
@@ -529,43 +529,49 @@ int ss_statsMerge( ShipStats *dest, const ShipStats *src )
  */
 int ss_statsMergeSingle( ShipStats *stats, const ShipStatList *list )
 {
-   char *ptr;
-   char *fieldptr;
-   double *dbl;
-   int *i;
-   const ShipStatsLookup *sl = &ss_lookup[ list->type ];
-
-   ptr = (char*) stats;
-   switch (sl->data) {
-      case SS_DATA_TYPE_DOUBLE:
-         fieldptr = &ptr[ sl->offset ];
-         memcpy(&dbl, &fieldptr, sizeof(double*));
-         *dbl *= 1.0+list->d.d;
-         if (*dbl < 0.) /* Don't let the values go negative. */
-            *dbl = 0.;
-         break;
-
-      case SS_DATA_TYPE_DOUBLE_ABSOLUTE:
-      case SS_DATA_TYPE_DOUBLE_ABSOLUTE_PERCENT:
-         fieldptr = &ptr[ sl->offset ];
-         memcpy(&dbl, &fieldptr, sizeof(double*));
-         *dbl += list->d.d;
-         break;
-
-      case SS_DATA_TYPE_INTEGER:
-         fieldptr = &ptr[ sl->offset ];
-         memcpy(&i, &fieldptr, sizeof(int*));
-         *i   += list->d.i;
-         break;
-
-      case SS_DATA_TYPE_BOOLEAN:
-         fieldptr = &ptr[ sl->offset ];
-         memcpy(&i, &fieldptr, sizeof(int*));
-         *i    = 1; /* Can only set to true. */
-         break;
-   }
+   return ss_statsMergeSingleScale(stats, list, 1);
```

Comment:
I'm pretty sure this function was previously just a copy-pasta of `ss_statsMergeSingleScale` with `scale = 1` so I've just called it directly here instead of making the changes below in two places. Can you check I haven't missed how these functions are in some way subtlety different?